.github/workflows/pr-check-lint_content.yml

```
@@ -0,0 +1,141 @@
+          git config --global user.name "mdn-bot"
+
+          rm -r files *.md
+          mv pr_head/files pr_head/*.md .
+          rm -r pr_head
+          git commit -am "Code from PR head"
+
+      - name: Get changed files
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          BASE_SHA: ${{ github.event.pull_request.base.sha }}
+          HEAD_SHA: ${{ github.event.pull_request.head.sha }}
+        run: |
+          # Use the GitHub API to get the list of changed files
+          # documentation: https://docs.github.com/rest/commits/commits#compare-two-commits
+          DIFF_DOCUMENTS=$(gh api repos/{owner}/{repo}/compare/${{ env.BASE_SHA }}...${{ env.HEAD_SHA }} \
+            --jq '.files | .[] | select(.status|IN("added", "modified", "renamed", "copied", "changed")) | .filename')
+          # filter out files that are not markdown
+          DIFF_DOCUMENTS=$(echo "${DIFF_DOCUMENTS}" | egrep -i "^files/.*\.md$" | xargs)
+          echo "DIFF_DOCUMENTS=${DIFF_DOCUMENTS}" >> $GITHUB_ENV
+
+      - name: Setup Node.js environment
+        uses: actions/setup-node@v3
+        with:
+          node-version-file: ".nvmrc"
+          cache: yarn
+
+      - name: Install all yarn packages
+        run: yarn --frozen-lockfile
+        env:
+          # https://github.com/microsoft/vscode-ripgrep#github-api-limit-note
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Lint and format markdown files
+        run: |
+          # Generate random delimiter
+          # https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#multiline-strings
+          EOF="$(openssl rand -hex 8)"
+
+          files_to_lint="${{ env.DIFF_DOCUMENTS }}"
+
+          echo "Running markdownlint --fix"
+          MD_LINT_FAILED=false
+          MD_LINT_LOG=$(yarn markdownlint-cli2 --fix ${files_to_lint} 2>&1) || MD_LINT_FAILED=true
+          echo "MD_LINT_LOG<<${EOF}" >> $GITHUB_ENV
+          echo "${MD_LINT_LOG}" >> $GITHUB_ENV
+          echo "${EOF}" >> $GITHUB_ENV
+          echo "MD_LINT_FAILED=${MD_LINT_FAILED}" >> $GITHUB_ENV
+
+          echo "Linting front-matter"
+          FM_LINT_FAILED=false
+          FM_LINT_LOG=$(node scripts/front-matter_linter.js --fix true ${files_to_lint} 2>&1) || FM_LINT_FAILED=true
+          echo "FM_LINT_LOG<<${EOF}" >> $GITHUB_ENV
+          echo "${FM_LINT_LOG}" >> $GITHUB_ENV
+          echo "${EOF}" >> $GITHUB_ENV
+          echo "FM_LINT_FAILED=${FM_LINT_FAILED}" >> $GITHUB_ENV
+
+          echo "Running Prettier"
+          yarn prettier -w ${files_to_lint}
+
+          if [[ -n $(git diff) ]]; then
+            echo "FILES_MODIFIED=true" >> $GITHUB_ENV
+          fi
+
+          # info for troubleshooting
+          echo MD_LINT_FAILED=${MD_LINT_FAILED}
+          echo FM_LINT_FAILED=${FM_LINT_FAILED}
+          git diff
+
+      - name: Setup reviewdog
+        uses: reviewdog/action-setup@v1
+        with:
+          reviewdog_version: latest
+
```

I think we still need this, see: yin1999/translated-content#24

```suggestion

      - name: Replace OWNER ID
        run: |
          # Because reviewdog needs the write permission to submit suggestions,
          # and it simply checks if the OWNER ID of the PR's HEAD and the repository
          # are the same, or if the branch is not from a forked repository.
          # Unfortunately, there is no way to check the write permission
          # for `secrets.GITHUB_TOKEN` in the workflow. And reviewdog still couldn't
          # determine the write permission even if the workflow is triggered by
          # the PR from the forked repository, https://github.com/reviewdog/reviewdog/issues/1210.
          #
          # As a workaround, replace those OWNER IDs with `1`.
          EVENT=$(jq '.pull_request.head.repo.owner.id = 1 | .pull_request.base.repo.owner.id = 1' "$GITHUB_EVENT_PATH")
          echo "${EVENT}" > "$GITHUB_EVENT_PATH"

```

---

Instead of using IDs to fail the readonly check, here we are targeting the next line in [the function HasReadOnlyPermissionGitHubToken](https://github.com/reviewdog/reviewdog/blob/0982b5e7d99640cb55f1a849390b24f9d48cfcbe/cienv/github_actions.go#L119C2-L119C66):
```js
return isForkedRepo && event.ActionName != "pull_request_target"
```
As this workflow is triggered using `pull_request_target` the function should return false.

---

Yes, I've ignored the event target. Thanks for clarifying this.

> @yin1999 if you commit this workflow in your fork then I'll submit a PR to test. :)

Of course. Let's test it first :)

---

> Of course. Let's test it first :)

The testing was successful https://github.com/yin1999/content/pull/1